src/webrtc/vad.md

```diff
@@ -28,41 +53,171 @@ When the use of it is negotiated, it indicates whether the encoder believes the
 Hopefully, you now have some understanding of the value in the `level` field, and we can jump right into the algorithm.
 Don't worry, it's not difficult at all.
 
-## Audio Level processing
+### Audio Level processing
+
 You could use the flag bit "V" if available, but we don't recommend using it for the production environment for 2 reasons:
+
 1. It is optional, so you cannot be sure it will always be there.
 2. Implementation isn't standardized, meaning that you can have inconsistent behavior depending on the sender.
 
-It is therefore worth considering implementing your algorithm based purely on the `level` field.
+So it's time to implement a different approach...
+
+## The algorithm
+
+First let's define the inputs and outputs of our function that would perform VAD. The function takes audio noise levels and the noise threshold value as inputs. It returns an information of whether the given audio levels indicate if the person is speaking based on the given threshold.
+
+![vad_func](assets/vad_func.png)
```

Comment:
Could we export images without backgrounds and make sure they are readable in different color themes?

Reply:
Since some of the diagrams are detailed and I couldn't find a color that would nicely fit all of the color themes, I have left the background as it was